### ISSUE TITLE
Several errors when running 'make all' for Hikey Board.

### ISSUE THREAD

**Issue by targetnull** [OP-TEE/optee_os#544]
Hello, I bought a Hikey Board a month ago and I want to run OP-TEE on it.
I use `repo` to get it, but I come across several errors as follows:

1. [My last issue, which turned out to be a bug](https://github.com/OP-TEE/optee_os/issues/542#issuecomment-155009384);
2. Apart from 1, it says `status=none` is not a supported flag, so I updated my `coreutils` to the latest stable version 8.24, and it has been **solved** also;
3. Below are the errors that sticks me right now. They happened after the two errors above were fixed. Specifically, it says that when making [**[boot-img]**

>sudo mkfs.fat -n "BOOT IMG" /home/targetnull/devel/optee_hikey/build/../out/boot-fat.uefi.img >/dev/null
sudo: mkfs.fat: command not found
make: *** [boot-img] Error 1

I only find **mkfs.vfat** instead of **mkfa.fat**, so I replaced it with `mkfa.vfat`, but another error just showed up!
>python gen_loader.py -o ptable.img --img_prm_ptable=prm_ptable.img --img_sec_ptable=sec_ptable.img
+-----------------------------------------------------------+
 Input Images:
     primary partition table:       prm_ptable.img
     secondary partition table:     sec_ptable.img
 Ouput Image:                       ptable.img
+-----------------------------------------------------------+
It's not partition table image.
make[1]: *** [ptable.img] Error 4
make[1]: Leaving directory `/home/targetnull/devel/optee_hikey/l-loader'
make: *** [lloader] Error 2

I did some google work but no luck:(. These errors seem to be so strange. My teammates have not come across these errors, but the only difference between us is the version of Ubuntu(me 12.04.5 and them 14.04.1), and I don't think this is the reason.

Could someone please explain a little about what I did wrong? Should I preinstall some libs or tools, or update some necessary software to latest version?

Thanks a lot
Tgn

**Comment by vchong** [comment]
Try reinstalling dosfstools.

**Comment by targetnull** [comment]
@vchong
Do you mean install with source code？ I installed it with apt-get.

**Comment by vchong** [comment]
No. Reinstall the dosfstools package and run ```which mkfs.fat```.
Btw, did you make any changes to the build script? It's mkfs.XXX, not mkfa.XXX. Make sure you don't have a typo.

**Comment by vchong** [comment]
Also, if your teammates have no errors and the only difference is the version of Ubuntu, then that is more than likely to be source of the reason for the failure. It's likely a problem with the build environment and not the build script.

**Comment by jbech-linaro** [comment]
Hi @targetnull 

I've tried myself to build this from scratch this morning and I couldn't see __any__ build issues. What I did was:
```
repo init -u https://github.com/OP-TEE/manifest.git -m hikey.xml
repo sync -j3
make toolchains
make all
```

When I'm about to flash the device and run:
```
make flash
```

I had to make a manual update in `hisi-idt.py`, since my computer doesn't figure out the chinese characters found in this script (and line)
https://github.com/96boards/burn-boot/blob/master/hisi-idt.py#L228

I basically just replace it with the following:
```
dev  = '/dev/serial/by-id/usb-FTDI_TTL232RG-VREG1V8_FTXOEBKH-if00-port0'
```

I saw some issues when booting (kernel crash), but the process of getting the source code and build is working on latest. So, as @vchong said, it is most likely a local setup problem on your own computer. Also, there are tons of people running this setup, so I doubt a problem in the scripts.

Have you installed all dependencies stated in https://github.com/OP-TEE/optee_os/README.md 4.1?
